src/services/platform.ts

```diff
@@ -12,9 +12,13 @@ async function matchTxToPlatform(tx: string): Promise<IPlatform> {
     const to = parsed.to.toLowerCase();
     const data = parsed.data.toLowerCase();
     const platforms = await Platform.find();
-
     for (const platform of platforms) {
-      for (const contract of platform.whitelist) {
+      const platformArbitrum = platform.whitelist.ethereum[parsed.chainId];
```

Comment:
Why is this still called `platformArbitrum`? You keep misnaming things which will lead to very confusing code so you need to really pay attention to this.

Reply:
I am now using Assetype.Ethereum. 
the chainId must be 42161, for community credits to deduct. In other cases it will give undefined platform making the platform credits nondeductible. Which is something we want i think.
the assetype must be ethereum as well. if not then the platfrom credits will be unreachable.
This situation might only effect the xfai whitelist address as i mentioned in asana.

Reply:
You're still not understanding. This function is a generic function for all transactions from all asset types and all chains. How is this supposed to work for Polkadot if you hardcode Ethereum? The `assetType` must come from the tx itself so it can be properly matched, that's why I keep telling you that you need a new parameter.
And as also mentioned in asana, just migrate the xfai data structure to match the new model